enumerables.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  def my_each
+    size = count - 1
+    counter = 0
+    items = to_a
+    while counter <= size
+      if is_a? Range
+        items.length >= 0 && yield(items[counter])
+      elsif is_a? Hash
+        yield(items[counter][0], items[counter][1])
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+    self
+  end
+
+  def my_each_with_index()
+    counter = 0
+    size = length - 1
+    while counter <= size
+      yield(self[counter], counter)
+      counter += 1
+    end
+    self
+  end
+
+  def my_select
+    required = []
+    my_each do |r|
+      required.push(yield(r))
+    end
+    required
+  end
+
+  def my_all?
+    track_negative = 0
+    my_each do |t|
+      yield(t) == true ? true : track_negative += 1
+    end
+    !track_negative.positive?
+  end
+
+  def my_any?
+    yes = 0
+    my_each do |y|
+      yield(y) == true ? yes += 1 : false
+    end
+    yes.positive?
+  end
+
+  def my_none?
+    satisfies = 0
+    my_each do |chek|
+      yield(chek) == false ? true : satisfies += 1
+    end
+    satisfies.positive? ? false : true
+  end
+
+  def my_count
+    counter = 0
+    my_each do |el|
+      yield(el) ? counter += 1 : false
+    end
+    counter
+  end
+
+  def my_map
+    cont_arr = []
+    my_each do |cont|
+      cont_arr.push(yield(cont)) if yield(cont)
+    end
+    cont_arr
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] returns a new array with the results of running block once for every element in enum.
```ruby
array = Array.new(100){rand(0...9)}
array.my_map(&block) == array.map(&block) #true
```
- [ ] returns an Enumerator if no block is given
```ruby
array.my_map == Enumerator #true
```